In [14]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [15]:
!ls /content/drive/MyDrive/data

test  train  valid


In [ ]:
import os

# Birden fazla klasörün listesi
klasorler = [
    r"/content/drive/MyDrive/data/benign",
    r"/content/drive/MyDrive/data/malignant",
    r"/content/drive/MyDrive/data/normal"
]

for klasor in klasorler:
    for dosya in os.listdir(klasor):
        # Hem 'mask' geçecek hem de '.png' olacak
        if "mask" in dosya.lower() and dosya.lower().endswith(".png"):
            yol = os.path.join(klasor, dosya)
            os.remove(yol)
            print("Silindi:", yol)

print("Bitti.")


Silindi: /content/drive/MyDrive/data/benign/benign (111)_mask.png
Silindi: /content/drive/MyDrive/data/benign/benign (109)_mask.png
Silindi: /content/drive/MyDrive/data/benign/benign (110)_mask.png
Silindi: /content/drive/MyDrive/data/benign/benign (11)_mask.png
Silindi: /content/drive/MyDrive/data/benign/benign (101)_mask.png
Silindi: /content/drive/MyDrive/data/benign/benign (106)_mask.png
Silindi: /content/drive/MyDrive/data/benign/benign (102)_mask.png
Silindi: /content/drive/MyDrive/data/benign/benign (108)_mask.png
Silindi: /content/drive/MyDrive/data/benign/benign (105)_mask.png
Silindi: /content/drive/MyDrive/data/benign/benign (107)_mask.png
Silindi: /content/drive/MyDrive/data/benign/benign (104)_mask.png
Silindi: /content/drive/MyDrive/data/benign/benign (1)_mask.png
Silindi: /content/drive/MyDrive/data/benign/benign (10)_mask.png
Silindi: /content/drive/MyDrive/data/benign/benign (100)_mask.png
Silindi: /content/drive/MyDrive/data/benign/benign (100)_mask_1.png
Silindi: /co

In [ ]:
# Source - https://stackoverflow.com/a
# Posted by mcandre, modified by community. See post 'Timeline' for change history
# Retrieved 2025-11-22, License - CC BY-SA 3.0

newpath = r'/content/drive/MyDrive/data/train'
if not os.path.exists(newpath):
    os.makedirs(newpath)

newpath = r'/content/drive/MyDrive/data/test'
if not os.path.exists(newpath):
    os.makedirs(newpath)

newpath = r'/content/drive/MyDrive/data/valid'
if not os.path.exists(newpath):
    os.makedirs(newpath)




In [ ]:
newpath = r'/content/drive/MyDrive/data/train/benign'
if not os.path.exists(newpath):
    os.makedirs(newpath)

newpath = r'/content/drive/MyDrive/data/train/malignant'
if not os.path.exists(newpath):
    os.makedirs(newpath)

newpath = r'/content/drive/MyDrive/data/train/normal'
if not os.path.exists(newpath):
    os.makedirs(newpath)



newpath = r'/content/drive/MyDrive/data/test/benign'
if not os.path.exists(newpath):
    os.makedirs(newpath)

newpath = r'/content/drive/MyDrive/data/test/malignant'
if not os.path.exists(newpath):
    os.makedirs(newpath)

newpath = r'/content/drive/MyDrive/data/test/normal'
if not os.path.exists(newpath):
    os.makedirs(newpath)




newpath = r'/content/drive/MyDrive/data/valid/benign'
if not os.path.exists(newpath):
    os.makedirs(newpath)

newpath = r'/content/drive/MyDrive/data/valid/malignant'
if not os.path.exists(newpath):
    os.makedirs(newpath)

newpath = r'/content/drive/MyDrive/data/valid/normal'
if not os.path.exists(newpath):
    os.makedirs(newpath)

In [ ]:
import os
import shutil
import random

def resimleri_tasi(kaynak_yolu, hedef_train, hedef_test, hedef_val, oran_train=0.7, oran_test=0.15, oran_val=0.15):

    # Klasörlerin var olduğundan emin olalım, yoksa hata verir
    os.makedirs(hedef_train, exist_ok=True)
    os.makedirs(hedef_test, exist_ok=True)
    os.makedirs(hedef_val, exist_ok=True)

    # Sadece png dosyalarını al
    dosyalar = [f for f in os.listdir(kaynak_yolu) if f.lower().endswith('.png')]
    random.shuffle(dosyalar) # Listeyi karıştır

    toplam_dosya = len(dosyalar)

    # Sayıları hesapla
    sayi_train = int(toplam_dosya * oran_train)
    sayi_test = int(toplam_dosya * oran_test)
    # Kalanların hepsi validation olsun (sayı yuvarlama hatası olmasın diye)
    sayi_val = toplam_dosya - sayi_train - sayi_test

    print(f"Toplam resim: {toplam_dosya}")
    print(f"Train'e gidecek: {sayi_train}")
    print(f"Test'e gidecek: {sayi_test}")
    print(f"Valid'e gidecek: {sayi_val}")

    # --- DİLİMLEME VE TAŞIMA (Kritik Düzeltme Burası) ---

    # 1. Train (%0 - %70 arası)
    train_dosyalari = dosyalar[:sayi_train]
    for dosya in train_dosyalari:
        shutil.move(os.path.join(kaynak_yolu, dosya), os.path.join(hedef_train, dosya))

    # 2. Test (%70 - %85 arası) -> Kaldığımız yerden devam ediyoruz
    test_dosyalari = dosyalar[sayi_train : sayi_train + sayi_test]
    for dosya in test_dosyalari:
        shutil.move(os.path.join(kaynak_yolu, dosya), os.path.join(hedef_test, dosya))

    # 3. Validation (Geriye kalanlar)
    val_dosyalari = dosyalar[sayi_train + sayi_test :]
    for dosya in val_dosyalari:
        shutil.move(os.path.join(kaynak_yolu, dosya), os.path.join(hedef_val, dosya))

    print("Tüm taşıma işlemi başarıyla tamamlandı!")

# --- KULLANIM ---
kaynak = "/content/drive/MyDrive/data/benign"

# Dikkat: Değişken isimleri farklı olmalı
hedef_tr = "/content/drive/MyDrive/data/train/benign"
hedef_ts = "/content/drive/MyDrive/data/test/benign"
hedef_vl = "/content/drive/MyDrive/data/valid/benign"

resimleri_tasi(kaynak, hedef_tr, hedef_ts, hedef_vl)







Toplam resim: 437
Train'e gidecek: 305
Test'e gidecek: 65
Valid'e gidecek: 67
Tüm taşıma işlemi başarıyla tamamlandı!


In [ ]:
import os
import shutil
import random

def resimleri_tasi(kaynak_yolu, hedef_train, hedef_test, hedef_val, oran_train=0.7, oran_test=0.15, oran_val=0.15):

    # Klasörlerin var olduğundan emin olalım, yoksa hata verir
    os.makedirs(hedef_train, exist_ok=True)
    os.makedirs(hedef_test, exist_ok=True)
    os.makedirs(hedef_val, exist_ok=True)

    # Sadece png dosyalarını al
    dosyalar = [f for f in os.listdir(kaynak_yolu) if f.lower().endswith('.png')]
    random.shuffle(dosyalar) # Listeyi karıştır

    toplam_dosya = len(dosyalar)

    # Sayıları hesapla
    sayi_train = int(toplam_dosya * oran_train)
    sayi_test = int(toplam_dosya * oran_test)
    # Kalanların hepsi validation olsun (sayı yuvarlama hatası olmasın diye)
    sayi_val = toplam_dosya - sayi_train - sayi_test

    print(f"Toplam resim: {toplam_dosya}")
    print(f"Train'e gidecek: {sayi_train}")
    print(f"Test'e gidecek: {sayi_test}")
    print(f"Valid'e gidecek: {sayi_val}")

    # --- DİLİMLEME VE TAŞIMA (Kritik Düzeltme Burası) ---

    # 1. Train (%0 - %70 arası)
    train_dosyalari = dosyalar[:sayi_train]
    for dosya in train_dosyalari:
        shutil.move(os.path.join(kaynak_yolu, dosya), os.path.join(hedef_train, dosya))

    # 2. Test (%70 - %85 arası) -> Kaldığımız yerden devam ediyoruz
    test_dosyalari = dosyalar[sayi_train : sayi_train + sayi_test]
    for dosya in test_dosyalari:
        shutil.move(os.path.join(kaynak_yolu, dosya), os.path.join(hedef_test, dosya))

    # 3. Validation (Geriye kalanlar)
    val_dosyalari = dosyalar[sayi_train + sayi_test :]
    for dosya in val_dosyalari:
        shutil.move(os.path.join(kaynak_yolu, dosya), os.path.join(hedef_val, dosya))

    print("Tüm taşıma işlemi başarıyla tamamlandı!")

# --- KULLANIM ---
kaynak = "/content/drive/MyDrive/data/normal"

# Dikkat: Değişken isimleri farklı olmalı
hedef_tr = "/content/drive/MyDrive/data/train/normal"
hedef_ts = "/content/drive/MyDrive/data/test/normal"
hedef_vl = "/content/drive/MyDrive/data/valid/normal"

resimleri_tasi(kaynak, hedef_tr, hedef_ts, hedef_vl)

Toplam resim: 133
Train'e gidecek: 93
Test'e gidecek: 19
Valid'e gidecek: 21
Tüm taşıma işlemi başarıyla tamamlandı!


In [ ]:
import os
import shutil
import random

def resimleri_tasi(kaynak_yolu, hedef_train, hedef_test, hedef_val, oran_train=0.7, oran_test=0.15, oran_val=0.15):

    # Klasörlerin var olduğundan emin olalım, yoksa hata verir
    os.makedirs(hedef_train, exist_ok=True)
    os.makedirs(hedef_test, exist_ok=True)
    os.makedirs(hedef_val, exist_ok=True)

    # Sadece png dosyalarını al
    dosyalar = [f for f in os.listdir(kaynak_yolu) if f.lower().endswith('.png')]
    random.shuffle(dosyalar) # Listeyi karıştır

    toplam_dosya = len(dosyalar)

    # Sayıları hesapla
    sayi_train = int(toplam_dosya * oran_train)
    sayi_test = int(toplam_dosya * oran_test)
    # Kalanların hepsi validation olsun (sayı yuvarlama hatası olmasın diye)
    sayi_val = toplam_dosya - sayi_train - sayi_test

    print(f"Toplam resim: {toplam_dosya}")
    print(f"Train'e gidecek: {sayi_train}")
    print(f"Test'e gidecek: {sayi_test}")
    print(f"Valid'e gidecek: {sayi_val}")

    # --- DİLİMLEME VE TAŞIMA (Kritik Düzeltme Burası) ---

    # 1. Train (%0 - %70 arası)
    train_dosyalari = dosyalar[:sayi_train]
    for dosya in train_dosyalari:
        shutil.move(os.path.join(kaynak_yolu, dosya), os.path.join(hedef_train, dosya))

    # 2. Test (%70 - %85 arası) -> Kaldığımız yerden devam ediyoruz
    test_dosyalari = dosyalar[sayi_train : sayi_train + sayi_test]
    for dosya in test_dosyalari:
        shutil.move(os.path.join(kaynak_yolu, dosya), os.path.join(hedef_test, dosya))

    # 3. Validation (Geriye kalanlar)
    val_dosyalari = dosyalar[sayi_train + sayi_test :]
    for dosya in val_dosyalari:
        shutil.move(os.path.join(kaynak_yolu, dosya), os.path.join(hedef_val, dosya))

    print("Tüm taşıma işlemi başarıyla tamamlandı!")

# --- KULLANIM ---
kaynak = "/content/drive/MyDrive/data/malignant"

# Dikkat: Değişken isimleri farklı olmalı
hedef_tr = "/content/drive/MyDrive/data/train/malignant"
hedef_ts = "/content/drive/MyDrive/data/test/malignant"
hedef_vl = "/content/drive/MyDrive/data/valid/malignant"

resimleri_tasi(kaynak, hedef_tr, hedef_ts, hedef_vl)

Toplam resim: 210
Train'e gidecek: 147
Test'e gidecek: 31
Valid'e gidecek: 32
Tüm taşıma işlemi başarıyla tamamlandı!




1.   Normal(93)	2325 için	Her resmi 25 kez çoğaltacağız
2.   Malignant(147)	2499 için	Her resmi 17 kez çoğaltacağız
3.   Benign(305)	2440 için	Her resmi 8 kez çoğaltacağız



In [16]:
!ls /content/drive/MyDrive/data/train/normal/

'normal (101).png'  'normal (14).png'  'normal (41).png'  'normal (78).png'
'normal (103).png'  'normal (15).png'  'normal (42).png'  'normal (79).png'
'normal (104).png'  'normal (16).png'  'normal (43).png'  'normal (7).png'
'normal (105).png'  'normal (18).png'  'normal (46).png'  'normal (80).png'
'normal (107).png'  'normal (19).png'  'normal (47).png'  'normal (82).png'
'normal (108).png'  'normal (1).png'   'normal (50).png'  'normal (83).png'
'normal (10).png'   'normal (21).png'  'normal (52).png'  'normal (84).png'
'normal (110).png'  'normal (22).png'  'normal (54).png'  'normal (85).png'
'normal (112).png'  'normal (23).png'  'normal (55).png'  'normal (86).png'
'normal (113).png'  'normal (24).png'  'normal (56).png'  'normal (87).png'
'normal (115).png'  'normal (25).png'  'normal (57).png'  'normal (88).png'
'normal (117).png'  'normal (26).png'  'normal (58).png'  'normal (89).png'
'normal (118).png'  'normal (28).png'  'normal (5).png'   'normal (8).png'
'normal (119).

In [20]:
from tensorflow.keras.preprocessing.image import ImageDataGenerator, load_img, img_to_array
import os

image_gen = ImageDataGenerator(rotation_range=15,
                               brightness_range=[0.8,1.2],
                               width_shift_range=0.10,
                               height_shift_range=0.10,
                               #rescale=1/255,
                               shear_range=0.1,
                               zoom_range=0.1,
                               horizontal_flip=True,
                               vertical_flip=False,
                               fill_mode='nearest')

for image in os.listdir('/content/drive/MyDrive/data/train/normal'):
  if not image.lower().endswith(('.png', '.jpg', '.jpeg')):
        continue
  try:
    loaded_image = load_img('/content/drive/MyDrive/data/train/normal/' + image, target_size=(224,224))
    x = img_to_array(loaded_image)
    x = x.reshape((1,) + x.shape)

    i = 0
    for batch in image_gen.flow(x, batch_size=1, save_to_dir='/content/drive/MyDrive/data/train/normal', save_prefix='aug', save_format='png'):
        i += 1
        if i >= 25:
            break
  except Exception as e:
        print(f"Bir resimde hata oldu, pas geçiliyor: {image} - Hata: {e}")







In [21]:
i=0
for image in os.listdir('/content/drive/MyDrive/data/train/normal'):
  i+=1
print(i)


2181


In [22]:
from tensorflow.keras.preprocessing.image import ImageDataGenerator, load_img, img_to_array
import os

image_gen = ImageDataGenerator(rotation_range=15,
                               brightness_range=[0.8,1.2],
                               width_shift_range=0.10,
                               height_shift_range=0.10,
                               #rescale=1/255,
                               shear_range=0.1,
                               zoom_range=0.1,
                               horizontal_flip=True,
                               vertical_flip=False,
                               fill_mode='nearest')

for image in os.listdir('/content/drive/MyDrive/data/train/benign'):
  if not image.lower().endswith(('.png', '.jpg', '.jpeg')):
        continue
  try:
    loaded_image = load_img('/content/drive/MyDrive/data/train/benign/' + image, target_size=(224,224))
    x = img_to_array(loaded_image)
    x = x.reshape((1,) + x.shape)

    i = 0
    for batch in image_gen.flow(x, batch_size=1, save_to_dir='/content/drive/MyDrive/data/train/benign', save_prefix='aug', save_format='png'):
        i += 1
        if i >= 8:
            break
  except Exception as e:
        print(f"Bir resimde hata oldu, pas geçiliyor: {image} - Hata: {e}")

In [23]:
i=0
for image in os.listdir('/content/drive/MyDrive/data/train/benign'):
  i+=1
print(i)


2471


In [24]:
from tensorflow.keras.preprocessing.image import ImageDataGenerator, load_img, img_to_array
import os

image_gen = ImageDataGenerator(rotation_range=15,
                               brightness_range=[0.8,1.2],
                               width_shift_range=0.10,
                               height_shift_range=0.10,
                               #rescale=1/255,
                               shear_range=0.1,
                               zoom_range=0.1,
                               horizontal_flip=True,
                               vertical_flip=False,
                               fill_mode='nearest')

for image in os.listdir('/content/drive/MyDrive/data/train/malignant'):
  if not image.lower().endswith(('.png', '.jpg', '.jpeg')):
        continue
  try:
    loaded_image = load_img('/content/drive/MyDrive/data/train/malignant/' + image, target_size=(224,224))
    x = img_to_array(loaded_image)
    x = x.reshape((1,) + x.shape)

    i = 0
    for batch in image_gen.flow(x, batch_size=1, save_to_dir='/content/drive/MyDrive/data/train/malignant', save_prefix='aug', save_format='png'):
        i += 1
        if i >= 17:
            break
  except Exception as e:
        print(f"Bir resimde hata oldu, pas geçiliyor: {image} - Hata: {e}")

In [25]:
i=0
for image in os.listdir('/content/drive/MyDrive/data/train/malignant'):
  i+=1
print(i)


2379


In [26]:
import shutil


klasor_yolu = '/content/drive/MyDrive/data'
hedef_yol = '/content/drive/MyDrive/ara_proje_data'
shutil.make_archive(hedef_yol, 'zip', klasor_yolu)


'/content/drive/MyDrive/ara_proje_data.zip'